src/fairness/print_hierarchy/print_hierarchy.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2020 Lawrence Livermore National Security, LLC
+(c.f. AUTHORS, NOTICE.LLNS, COPYING)
+
+This file is part of the Flux resource manager framework.
+For details, see https://github.com/flux-framework.
+
+SPDX-License-Identifier: LGPL-3.0
+*/
+#include <iostream>
+#include <sqlite3.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <vector>
+#include <tuple>
+
+std::string hierarchy = "Bank|User|RawShares\n";
+std::string indent = "";
+char* messageError;
+
+sqlite3_stmt *b_select_root_bank_stmt;
+sqlite3_stmt *b_select_shares_stmt;
+sqlite3_stmt *b_select_sub_banks_stmt;
+sqlite3_stmt *b_select_associations_stmt;
```

Comment:
For a self-contained file, I think having these as globals is fine. If we want to expose `get_sub_banks` as a function that other parts of the codebase can call, then I think these should be arguments that are declared in `main` (or a public-facing API function, see note below on `main`) and then passed down into `get_sub_banks`.